### ISSUE TITLE
Which measure to use for gene Fold change analysis between pre- and post-treatment samples and for gene comparison within a sample

### ISSUE THREAD

**Issue by mpunta** [mikelove/tximport#49]
Hi, new to bulkRNAseq analysis and a bit confused about different normalization measures that are used.

My understanding is that TPMs are fine for comparison of gene expression levels within a sample (so also for looking at correlation of expression between two genes in many samples separately) while for looking at fold change expression of list of genes of interest in two samples (e.g. pre and post treatment) the way to go is to transform the estimated counts from e.g. kallisto (imported with tximport using the CountsfromAbindance = "no" option) with the edgeR pipeline described in the tximport vignette. What confuses me is:
1) what is obtained at the end of the edgeR code described in the vignette are called CPMs (cpms <- edgeR::cpm(y, offset = y$offset, log = FALSE) ) while to me these appear very similar to TMMs instead (since they are calculated accounting both for different average transcript length between samples and for composition biases between samples). Am I getting this wrong?

2) the vignette appears to suggest that processing the data with the estimated ("raw") counts with the edgeR pipeline has a similar effect as using scaledTPM or lengthscaledTPM but in fact what is obtained are very different counts. Quoting from one of your answer in (https://support.bioconductor.org/p/84883/): "These two have library size differences baked in. The column sum is equal to the number of mapped reads. So not comparable across samples." While as per 1) above I believe that the "CPMs" obtained with the edgeR vignette CAN be used for across sample comparison. Am I right?

In conclusion, what measure would you suggest to use for calculating logFC of a list of genes of interest between a pre and a post treatment sample? Is it the CPMs as obtained via the edgeR pipeline in the vignette or something else? And if the first, I guess those "CMPs" could be used also for looking at how pairs of genes correlate in different samples (one point per sample in a correlation plot)

Thank you very much in advance for any help tyou will ge able to give me.

Marco

**Comment by mikelove** [comment]
If you have follow up questions, let's continue on the support site, as this is where most people go to look for tximport usage questions.

_"the way to go is ... the estimated counts...with the edgeR pipeline described in the tximport vignette"_

Yes, the default recommendation is to import estimated counts and use the gene lengths as a per-sample offset. This is what happens if you use the DESeq2 or edgeR code in the vignette.

For CPM vs TMM-scaling, you'll have to consult the edgeR vignette.

_"...suggest that processing the data with the estimated ("raw") counts with the edgeR pipeline has a similar effect as using scaledTPM or lengthscaledTPM"_ 

No, raw counts alone are not similar to scaledTPM or lengthScaledTPM. It is the count + offset that gives a similar result, not count analysis alone.

_"what measure would you suggest to use for calculating logFC"_

The LFC estmated by DESeq2 or edgeR and returned in the results table or topTable after using any of the tximport recommended pipelines (estimated count + offset, or the counts-from-abundance option).
